src/components/ArticleCard.jsx

```diff
@@ -1,14 +1,39 @@
+import { useState } from "react";
+import axios from "axios";
 
 export default function ArticleCard({ article }) {
+	const [votes, setVotes] = useState(article.votes);
+	const [isError, setIsError] = useState(false);
+	console.log(votes);
+
+	const handleVote = (voteChange) => {
+		setVotes(votes + voteChange); // Optimistic change in votes
+		axios
+			.patch(`https://news-project-baar.onrender.com/api/articles/${article.article_id}`, {
+				inc_votes: voteChange
+			})
+			.then(({ data }) => {
+				console.log(data.article.votes);
+				setVotes(data.article.votes);
+			})
+			.catch((err) => {
+				console.error('Error updating the vote. Please try again: ', err);
+				setVotes(votes - voteChange);
+				setIsError(true)
```

Comment:
I can see the error message in the console if i switch my network to offline. Having some kind of error message display on the page for the user would make it nicer for the user to see if their vote failed